### PR TITLE
Make task for devel and kint integration

### PR DESCRIPTION
**Pull request by xArturia**
#17 
Devel + kint setup and config for twig debugging (using sed).

Inside the devel task, the command "drush pm-uninstall" is ignoring error exit codes to avoid it breaking up the build
when the module is not found, example:

```dynamic_page_cache is already uninstalled.                                                                                                        [ok]
Extension internal_page_cache was not found and will not be uninstalled.                                                                     [warning]
There were no extensions that could be uninstalled.                                                                                               [ok]
Makefile:97: recipe for target 'devel' failed
make: [devel] Error 129 (ignored)```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,3 +75,29 @@ front:
 
 iprange:
 	$(shell grep -q -F 'IPRANGE=' .env || echo "\nIPRANGE=$(shell docker network inspect $(COMPOSE_PROJECT_NAME)_front --format '{{(index .IPAM.Config 0).Subnet}}')" >> .env)
+
+devel:
+	@echo "Setting up permissions ..."
+	sudo find build/sites/default -type d -exec chmod 777 {} \;
+	sudo find build/sites/default -type f -exec chmod 666 {} \;
+
+	docker-compose -p farc exec php find sites/default -type d -exec chmod 777 {} \;
```
Comment (andypost) [inline Makefile]: please remove `-p farc` all over, project used from `.env`

File: Makefile [hunk 2]
```diff
@@ -75,3 +75,29 @@ front:
 
 iprange:
 	$(shell grep -q -F 'IPRANGE=' .env || echo "\nIPRANGE=$(shell docker network inspect $(COMPOSE_PROJECT_NAME)_front --format '{{(index .IPAM.Config 0).Subnet}}')" >> .env)
+
+devel:
+	@echo "Setting up permissions ..."
+	sudo find build/sites/default -type d -exec chmod 777 {} \;
+	sudo find build/sites/default -type f -exec chmod 666 {} \;
```
Comment (andypost) [inline Makefile]: better beware `sudo` with `docker-compose exec php ...`

File: Makefile [hunk 3]
```diff
@@ -75,3 +75,29 @@ front:
...
+
+	@echo "Setting up settings.yml ..."
+	docker-compose -p farc exec php cp sites/default/default.services.yml sites/default/services.yml
+
+	@echo "Setting up settings.local.yml ..."
+	docker-compose -p farc exec php cp sites/example.settings.local.php sites/default/settings.local.php
+
+	@echo "Setting up kint ..."
+	-docker-compose -p farc exec php drush en devel devel_generate kint -y
+	-docker-compose -p farc exec php drush pm-uninstall dynamic_page_cache internal_page_cache -y
```
Comment (andypost) [inline Makefile]: what `-` (minus) means?

**Comment by xArturia** [comment]
The - are for drush, to avoid exit code differents to 0 so build doesn't fail (like when trying to uninstall a module that isn't installed)

**Comment by andypost** [comment]
Looks good to me

File: README.md [hunk 4]
```diff
@@ -56,3 +56,4 @@ For Linux install <a href="https://docs.docker.com/compose/install/" target="_bl
 * `make info` - Show project info (IP).
 * `make chown` - Change permissions inside container. Use it in case you can not access files in _build_. folder from your machine.
 * `make exec` - docker exec into php container.
+* `make exec` - Devel + kint setup, and config for Twig debug mode.
```
Comment (andriyun) [inline README.md]: type `make devel` need

**Comment by andriyun** [comment]
Looks ok for me too safe small notice above

File: Makefile [hunk 5]
```diff
@@ -75,3 +75,26 @@ front:
...
+	docker-compose exec php find sites/default -type d -exec chmod 777 {} \;
+	docker-compose exec php find sites/default -type f -exec chmod 666 {} \;
+
+	@echo "Setting up settings.yml ..."
+	docker-compose exec php cp sites/default/default.services.yml sites/default/services.yml
+
+	@echo "Setting up settings.local.yml ..."
+	docker-compose exec php cp sites/example.settings.local.php sites/default/settings.local.php
+	@echo "Setting up kint ..."
+	-docker-compose exec php drush en devel devel_generate kint -y
```
Comment (gaydabura) [inline Makefile]: need to download devel before, or add notice to include in make.yml

Comment (xArturia) [inline Makefile]: Added composer require for devel

**Comment by xArturia** [comment]
Updated README, added composer require for devel, and applied fix for permissions issues after updating container (chown task).

**Comment by andriyun** [comment]
Looks ok for me

**Pull request merged**